src/py2spack/core.py

```diff
@@ -342,79 +312,91 @@ class SpackPyPkg:
-        self.pypi_name: str = ""
-        self.description: str | None = None
-        self.pypi: str = ""
-        self.versions: list[tuple[sv.Version, str]] = []
-        self.all_versions: list[pv.Version]
-        self.variants: set[str] = set()
-        self.maintainers: list[str] = []
-        self.authors: list[str] = []
-        self.license: str | None = None
-        self.homepage: str | None = None
-        self.dependencies_by_type: dict[str, list[tuple[spec.Spec, spec.Spec]]] = {}
-        self.file_parse_errors: list[tuple[str, ParseError]] = []
-        self.metadata_parse_errors: dict[str, list[parsing.ConfigurationError]] = {}
-        self.dependency_parse_errors: dict[str, list[parsing.ConfigurationError]] = {}
-        self.dependency_conversion_errors: dict[str, list[conversion_tools.ConversionError]] = {}
+        self._name: str = ""
+        self._pypi_name: str = ""
+        self._description: str | None = None
+        self._pypi: str = ""
+        self._versions_with_checksum: list[tuple[sv.Version, str, str]] = []
+        self._versions_missing_checksum: list[sv.Version] = []
+        self._all_versions: list[pv.Version]
+        self._variants: set[str] = set()
+        self._maintainers: list[str] = []
+        self._authors: list[str] = []
+        self._license: str | None = None
+        self._homepage: str | None = None
+        self._dependencies_by_type: dict[str, list[tuple[spec.Spec, spec.Spec]]] = {}
+        self._file_parse_errors: list[tuple[str, ParseError]] = []
+        self._metadata_parse_errors: dict[str, list[pyproject_parsing.ConfigurationError]] = {}
+        self._dependency_parse_errors: dict[str, list[pyproject_parsing.ConfigurationError]] = {}
+        self._dependency_conversion_errors: dict[str, list[conversion_tools.ConversionError]] = {}
         # map each unique dependency (dependency spec, when spec) to a
         # list of package versions that have this dependency
-        self.specs_to_versions: dict[tuple[spec.Spec, spec.Spec], list[pv.Version]] = {}
+        self._specs_to_versions: dict[tuple[spec.Spec, spec.Spec], list[pv.Version]] = {}
         # map dependencies to their dependency types (build, run, test, ...)
-        self.specs_to_types: dict[tuple[spec.Spec, spec.Spec], set[str]] = {}
+        self._specs_to_types: dict[tuple[spec.Spec, spec.Spec], set[str]] = {}
 
     @staticmethod
     def convert_pkg(
-        name: str, lookup: loading.PyPILookup, last_n_versions: int = 20
+        name: str, provider: package_providers.PyProjectProvider, last_n_versions: int = 20
     ) -> SpackPyPkg | None:
         """Convert a PyPI package to a Spack package.py."""
-        # download metadata on available sdists from PyPI
-        files = lookup.get_files(name, last_n_versions=last_n_versions)
-        if isinstance(files, loading.APIError):
-            logging.warning("PyPI API Error: %s", str(files))
+        # download available versions through provider (pypi, github)
```

Comment:
Changed code to adapt new package_providers API. Download versions, later get the sdist files directly using the version.